client/src/main/java/io/pravega/client/segment/impl/AsyncSegmentInputStreamImpl.java

```diff
@@ -84,9 +85,21 @@ public void connectionDropped() {
             closeConnection(new ConnectionFailedException());
         }
 
+
         @Override
         public void wrongHost(WireCommands.WrongHost wrongHost) {
-            closeConnection(new ConnectionFailedException(wrongHost.toString()));
+            log.info("Received wrongHost {}", wrongHost);
+            ClientConnection conn;
+            try {
+                conn = connection.get();
```

Comment:
If the future is not complete this may be incorrect. It is probably better to check isSuccess (see Futures in common). Then you can do `.join()` which removes the need for the catch.

Reply:
Incorporated changes.